docs-src/security-model.md

```diff
@@ -0,0 +1,148 @@
+  GatewayClass.
+
+## Roles
+For the purposes of this security model, 3 common roles have been identified:
+
+* **Infrastructure provider**: The infrastructure provider (infra) is
+  responsible for the overall environment that the cluster(s) are operating in.
+  Examples include: the cloud provider (AWS, Azure, GCP, ...), the PaaS provider
+  in a company.
+* **Cluster operator**: The cluster operator (ops) is responsible for
+  administration of entire clusters. They manage policies, network access,
+  application permissions.
+* **Application developer**: The application developer (dev) is responsible for
+  defining their application configuration (e.g. timeouts, request
+  matching/filter) and Service composition (e.g. path routing to backends). 
+
+Although these roles can cover a wide variety of use cases, some organizations
+may be structured slightly differently. Many organizations may also have a
+fourth role that sits between "cluster operator" and "application developer":
+
+* **Application admin**: The application admin has administrative access to some
+  namespaces within a cluster, but not the cluster as a whole.
+
+## The Security Model
+There are two primary components to the Service APIs security model: RBAC and
+namespace restrictions.
+
+## RBAC
+RBAC (role-based access control) is the standard used for Kubernetes
+authorization. This allows users to configure who can perform actions on
+resources in specific scopes. RBAC can be used to enable each of the roles
+defined above. In most cases, it will be desirable to have all resources be
+readable by most roles, so instead we'll focus on write access for this model.
+
+### Write Permissions for Simple 3 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | No | Yes | Yes | Yes |
+| Application Developers | No | No | Yes | Yes |
+
+### Write Permissions for Advanced 4 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | Sometimes | Yes | Yes | Yes |
+| Application Admins | No | In Specified Namespaces | In Specified Namespaces | In Specified Namespaces |
+| Application Developers | No | No | In Specified Namespaces | In Specified Namespaces |
+
+## Namespace Restrictions
+The extra configuration options are not possible to control with RBAC. Instead,
+they will be controlled with configuration fields on GatewayClasses:
+
+* **gatewayNamespaceSelector**: This field is a selector of valid namespaces
+  that Gateways can use this GatewayClass from. An empty selector (default)
+  indicates that Gateways can use this GatewayClass from any namespace.
+
+* **routeNamespaceSelector**: This field is a selector of valid namespaces that
+  gateways can reference routes from. An empty selector (default) indicates that
+  routes can be referenced from any namespace.
```

Comment:
should this be secure by default(empty means can only reference routes within gateway namespaces)? i noticed it's in the doc(https://docs.google.com/document/d/14-mDATLLiF70wFaUPDXxFjO9BDrl9863Lk8um2RaTLs/edit) but changed here.

Reply:
Good catch, thanks! I've update this to include that the default (nil) value should be interpreted with the secure by default approach.